modules/azure/compute.go

```diff
@@ -10,15 +10,12 @@ import (
 
 // GetVirtualMachineClientE is a helper function that will setup an Azure Virtual Machine client on your behalf.
 func GetVirtualMachineClientE(subscriptionID string) (*compute.VirtualMachinesClient, error) {
```

Comment:
we do need a non-e method for this.  I double checked AWS impl and they follow that pattern even on client.

Reply:
Noting this was in the `compute` module; not anything related to the `client_factory`. But definitely will update to bring into spec.

Reply:
good point, let's not add the non-E method for now and see what gruntworks says.

Reply:
I added the non-E method to `compute.go` just so it followed the rest of our code .